src/client.rs

```diff
@@ -157,6 +170,48 @@ impl Client {
             bucket_name: bucket_name.to_owned(),
             collection_name: collection_name.to_owned(),
             verifier: verifier.unwrap_or_else(|| Box::new(AlwaysAcceptsVerifier{})),
+            cache: Box::new(DefaultCache{}),
+        };
+    }
+
+    fn fetch_records_with_verification(&self) -> Result<Vec<KintoObject>, ClientError> {
+
+        let remote_timestamp = self.get_latest_collection_timestamp()?;
+
+        let changeset_response = get_changeset(
+            &self.server_url,
+            &self.bucket_name,
+            &self.collection_name,
+            remote_timestamp,
+        )?;
+
+        debug!("changeset.metadata {}", serde_json::to_string_pretty(&changeset_response.metadata)?);
+
+        let collection: Collection = self.create_collection_from_response(changeset_response);
+        self.verifier.verify(&collection)?;
+
+        // before returning records, we want to override it into the cache
+        self.cache.store(&format!("{}/{}", collection.bid, collection.cid), json!(collection.records))?;
```

Comment:
The key should contain the type of data. For example `{bid}/{cid}:records`

Reply:
Also, I think the storing should be done in the same place as retrieving (even if it means to have repeated calls to `self.cache.store()`